table/src/main/java/org/enso/table/data/column/storage/BoolStorage.java

```diff
@@ -133,4 +121,138 @@ public Storage countMask(int[] counts, int total) {
+                  BoolStorage v = (BoolStorage) arg;
+                  BitSet missing = v.isMissing.get(0, storage.size);
+                  missing.or(storage.isMissing);
+                  BitSet out = v.values.get(0, storage.size);
+                  boolean negated;
+                  if (storage.negated && v.negated) {
+                    out.or(storage.values);
+                    negated = true;
+                  } else if (storage.negated) {
+                    out.andNot(storage.values);
+                    negated = false;
+                  } else if (v.negated) {
+                    out.flip(0, storage.size);
+                    out.and(storage.values);
+                    negated = false;
+                  } else {
+                    out.and(storage.values);
+                    negated = false;
+                  }
+                  return new BoolStorage(out, missing, storage.size, negated);
+                } else {
+                  throw new UnexpectedColumnTypeException("Boolean");
+                }
+              }
+            })
+        .add(
+            new MapOperation<>(Ops.OR) {
+              @Override
+              public Storage runMap(BoolStorage storage, Object arg) {
+                if (arg instanceof Boolean) {
+                  boolean v = (Boolean) arg;
+                  if (v) {
+                    return new BoolStorage(new BitSet(), storage.isMissing, storage.size, true);
+                  } else {
+                    return storage;
+                  }
+                } else {
+                  throw new UnexpectedTypeException("a Boolean");
+                }
+              }
+
+              @Override
+              public Storage runZip(BoolStorage storage, Storage arg) {
+                if (arg instanceof BoolStorage) {
+                  BoolStorage v = (BoolStorage) arg;
+                  BitSet missing = v.isMissing.get(0, storage.size);
+                  missing.or(storage.isMissing);
+                  BitSet out = v.values.get(0, storage.size);
+                  boolean negated;
+                  if (storage.negated && v.negated) {
+                    out.and(storage.values);
+                    negated = true;
+                  } else if (storage.negated) {
+                    out.flip(0, storage.size);
+                    out.and(storage.values);
+                    negated = true;
+                  } else if (v.negated) {
+                    out.flip(0, storage.size);
+                    out.or(storage.values);
+                    negated = false;
+                  } else {
+                    out.or(storage.values);
+                    negated = false;
+                  }
+                  return new BoolStorage(out, missing, storage.size, negated);
+                } else {
+                  throw new UnexpectedColumnTypeException("Boolean");
+                }
+              }
+            });
+    return ops;
```

Comment:
What do you think about adding `xor`? It's a boolean op I used fairly frequently back at Bloomberg for data stuff.

Reply:
But we have xor already! It's just that it's called `!=` :P